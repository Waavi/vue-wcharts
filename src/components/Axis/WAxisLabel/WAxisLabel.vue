<template>
    <text
        :x="x"
        :y="y"
        :text-anchor="textAnchor"
        :transform="transform"
    >
        <tspan
            v-for="(text, index) in value"
            :key="index"
            :x="x"
            :dy="`${index ? lineHeight : 0}em`"
            :style="styles"
        >
            {{ text }}
        </tspan>
    </text>
</template>

<script>
import VueTypes from 'vue-types'

export default {
    name: 'WAxisLabel',
    props: {
        x: VueTypes.number,
        y: VueTypes.number,
        textAnchor: VueTypes.string,
        transform: VueTypes.string,
        value: VueTypes.arrayOf(VueTypes.string),
        styles: VueTypes.any,
        lineHeight: VueTypes.number.def(1.25),
    },
}
</script>
