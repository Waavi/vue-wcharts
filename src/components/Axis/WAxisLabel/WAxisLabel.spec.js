import { shallowMount } from '@vue/test-utils'
import WAxisLabel from './WAxisLabel.vue'

describe('Common/WAxisLabel', () => {
    const propsData = {
        textAnchor: 'middle',
        transform: 'rotate(-90 30 135)',
        value: ['Production loss from potential', 'out of service (â‚¬/year)'],
        x: 30,
        y: 135,
        styles: {
            fontSize: '12px',
            color: '#444',
        },
    }

    const provide = {
        Chart: {
            active: {
                el: null,
                event: null,
            },
            activeElements: [],
            setActive: () => undefined,
            cleanActive: () => undefined,
        },
    }

    const defaultConfig = {
        propsData,
        provide,
    }

    it(`Should be render correctly`, () => {
        const wrapper = shallowMount(WAxisLabel, defaultConfig)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
