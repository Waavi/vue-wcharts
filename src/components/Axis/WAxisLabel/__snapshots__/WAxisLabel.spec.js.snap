// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common/WAxisLabel Should be render correctly 1`] = `
<text x="30" y="135" text-anchor="middle" transform="rotate(-90 30 135)">
  <tspan x="30" dy="0em" style="font-size: 12px; color: rgb(68, 68, 68);">
    Production loss from potential
  </tspan>
  <tspan x="30" dy="1.25em" style="font-size: 12px; color: rgb(68, 68, 68);">
    out of service (â‚¬/year)
  </tspan>
</text>
`;
